app/models/user.rb

```diff
@@ -1,8 +1,17 @@
 class User < ApplicationRecord
-  has_secure_password
+  has_secure_password validations: false, if: "uid"
+  validates :email, confirmation: true, if: "uid.nil?", on: :create
```

Comment:
Idea Add a test to make sure someone can't log in with an internal account with a particular email address, and then also creates an account with github associated with the same email address